src/main/scala/infrastructure/repositories/InMemoryCustomerRepository.scala

```diff
@@ -0,0 +1,33 @@
+package infrastructure.repositories
+
+import cats.effect.IO
+import cats.implicits.{catsSyntaxNestedFoldable, catsSyntaxParallelSequence_, toTraverseOps}
+import domain.{Customer, CustomerId}
+import domain.CustomerRepository.CustomerRepository
+
+class InMemoryCustomerRepository extends CustomerRepository[IO] {
```

Comment:
u cannot assume `IO` as an effect type, u need [F] as well, and InMemoryCustomerRepository needs to be F[_]

`MonadThrow` needs to be dragged bottom-up